### DISCUSSION TITLE
Any project plans in reaction to Nrwl taking over Lerna?

### DISCUSSION THREAD

**Discussion by kuhe** [ghiscoding/lerna-lite#158] [General]
https://github.com/lerna/lerna/issues/3121

and https://blog.nrwl.io/lerna-is-dead-long-live-lerna-61259f97dbd9

**Comment by ghiscoding** [comment]
Thanks for sharing the blog, I was waiting for it and didn't notice it.

No plan on the short term, when I created Lerna-Lite I was mainly focused on updating all dependencies and then eventually made it smaller (for my use case, I only ever used 3 commands, `publish`, `version` and `run`), so I was thinking why would I need to download the entire all-in-one Lerna lib with all 15 commands codebase? So I made it more modular to only include `publish` and `version`, all other commands are optional and this allow users to download less code and do roughly the same thing. The other thing that I recently added, 2 days ago, is the support for yarn/pnpm `workspace:` protocol which a lot of users begged for in the past 2 years... so with all of that in mind and the fact that Nrwl is taking over Lerna, the questions that I have (and that you might have to) are the following:
1. Will Lerna ever become more modular than it is today? 
   - because as it stands today, an all-in-one tool is no longer necessary since every package manager now have support for workspaces and I no longer want to download the entire Lerna codebase anymore
2. Will Lerna become a tool that only works with Nrwl toolset like Nx or will they keep it as a separate and independent tool?
   -  in my use case, I'd be more interested in moving to pnpm workspace and would be less interested in Nx, so hopefully Lerna will remain a standalone tool
    - EDIT: Lerna 5+ now requires TypeScript & Nx behind the scenes (Lerna-Lite does not), this is especially awful for Turborepo users.
3. When will Lerna support `workspace:` protocol that Lerna-Lite just brought support? 
   - I'm sure they will eventually but how long will that take?
      - EDIT: they have finally achieved it, 6 months later than Lerna-Lite 😉 
4. Can Lerna-Lite be merged into Lerna codebase?
   - No, it's written in TypeScript and Lerna is not, I also created the project from scratch, I copied every file 1 by 1 (so it's not technically a fork, even though I often say it is)
5. Will I personally ever go back to use the original Lerna and stop using my own Lerna-Lite fork?
   - I guess, not in the short term for sure and mostly because of number 1, Lerna-Lite is more smaller (Lite) and modular. On the long term, we'll see what happens with Lerna itself and I'm curious to see what will happen with number 2. 

In summary, I'm especially concerned about number 2 and I still prefer to use Lerna-Lite because of number 1. When I created Lerna-Lite, I was more interested in updating all dependencies since they were quite outdated at the time and I wasn't looking too much at the modularity but now that I have it, then I think it's a great advantage for its smaller size in comparison to Lerna.

Also, we'll see how often they'll change the codebase but as it is today, the code that is in Lerna-Lite came from Lerna, so I can still follow Lerna PRs and apply their changes into Lerna-Lite by replicating their PRs (when possible). Nrwl has a blog talking about computational caching which might be interesting but again we'll see how big of a change that will be on the codebase to implement it, if it's thousand of lines to change then that might be harder to replicate. 

I'm not sure if that addresses your question, but that's my take on it for the short term. At the end of the day, I'm just a regular developer/user of Lerna and I created Lerna-Lite to help my own project and if it's helping others (I think it does consider all the stars it has now) if that is the case, then it's an satisfying enough as a bonus

We'll see what happens next with Lerna in the future, time will tell.
Cheers and happy coding

### EDIT

I was afraid of number 2 and unfortunately, but not surprisingly, it came true. Lerna 5+ now has installs Nx and even TypeScript as required [dependencies](https://github.com/lerna/lerna/blob/main/core/lerna/package.json#L66-L67) (want it or not), which is not the case in Lerna-Lite (and **will never be**). So... Lerna-Lite is here to stay. :rocket:
- even if [Nx](https://nx.dev/) can be a nice addition with `lerna run` and [`useNx`](https://github.com/ghiscoding/lerna-lite/tree/main/packages/run#usenx-experimental) (which we did add as an option), it should and **will always be optional** in Lerna-Lite, it's your decision.

**Reply by ghiscoding** [reply]
> 2\. Will Lerna become a tool that only works with Nrwl toolset like Nx or will they keep it as a separate and independent tool?

As I suspected, number 2 has already started to be ~partially~ **totally** in play, it didn't take too long, with their latest version of Lerna [v5.2.0](https://github.com/lerna/lerna/releases/tag/v5.2.0), which is to install Nx for ~new~ **all users** and even though users could remove it, most user won't know that they could and this will just bring confusion thinking that Nx is a hard dependency that must always be installed at all time (which is not at all true). 

Of course this is all marketing stuff, but it's sad to see that they opted to go in that direction which I was afraid of... their only answer to this was "Nx brings a benefit since it runs faster so we should enable it by default", which is true that it runs faster with their caching system but it's not the only product that does such thing, TurboRepo os also another great alternative which does the same.

I'm not the only one thinking this is the wrong direction to go, @AviVahl wrote the same comment aloud which I had in the back of my mind 
> This installs `nx` and its many dependencies (some of which are requesting exact versions) to any project using lerna, even if they don't use `nx`. 👎
> Even stuff that I wouldn't expect like `tsconfig-paths` and `@parcel/watcher` are now being installed.

So in the end, I think that [Lerna-Lite](https://github.com/ghiscoding/lerna-lite) (this project), is here to stay and will be a great alternative to Lerna, Lerna-Lite is much smaller compared to Lerna and it's even more true considering latest Lerna changes. Note that I do want to keep PRs in sync with Lerna, but I will not add changes like this into Lerna-Lite to force push the `useNx` for new users, I did however bring the `useNx` as an optional feature and it should remain optional, if anyone want to install Nx, well they should chose for themselves instead of being force push into it.

**Reply by ghiscoding** [reply]
oh gosh it's actually worst than I imaged, I didn't realized it at first but just looking back at their latest release [v5.2.0](https://github.com/lerna/lerna/releases/tag/v5.2.0) and PR, Lerna is now, installing `Nx` as a dependency, want it or not, it is no longer optional at all, this is just 😱. I'm actually glad that I created Lerna-Lite before Nrwl took over Lerna and I'm totally keeping its use in my own repos... I'm sure there will be some backlash eventually when people realize that installing Lerna just got a whole lot bigger, at least a couple of persons already did realize that! Why haven't they considered this as a Breaking Change is just blowing my mind, this is very troublesome, so they seem to consider this as a Feature!?! Unless they just wanted to quietly push it to everyone without considering the possibility of advising anyone. 

See the Lerna [package.json](https://github.com/lerna/lerna/pull/3255/files#diff-8ef54c3d3a539622944e7149f253504b13854191a26207fb2cbede98ee0ef1a5) file change

Again not to worry with Lerna-Lite, I **will never apply this change** in Lerna-Lite. I can be picky in what PRs I decide to replicate back in here 😆 I did however replicate the `useNx` if anyone wants to use it, but it remains entirely **optional** like it should always be. I will not force push anything that should not be enforced in my own opinion.

Cheers and happy coding
